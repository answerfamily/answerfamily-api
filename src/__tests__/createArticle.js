jest.mock('../dataloaders/auth0UserLoaderFactory');

const gql = require('../../test/gql');
const esClient = require('../lib/esClient');
const mongoClient = require('../lib/mongoClient');

describe('createArticle', () => {
  it('connects reply and paragraph', async () => {
    const article = {
      text: '新文章ㄛ',
      paragraphs: [{ text: '新聞' }],
      sources: [{ url: 'http://google.com', note: '真的' }],
    };

    const { errors, data } = await gql`
      mutation($article: ArticleInput!) {
        createArticle(article: $article) {
          id
          text
          paragraphs {
            text
            user {
              id
            }
          }
          sources {
            url
            note
            user {
              id
            }
          }
          user {
            id
          }
        }
      }
    `(
      { article },
      {
        userPromise: Promise.resolve({ iss: 'user-id' }),
      }
    );

    expect(errors).toBeUndefined();

    const {
      // eslint-disable-next-line no-unused-vars
      createArticle: { id, ...newArticleData },
    } = data;
    expect(newArticleData).toMatchSnapshot();

    // cleanup
    await esClient.deleteByQuery({
      index: '*',
      body: {
        query: {
          match_all: {},
        },
      },
    });

    const { db } = await mongoClient;
    await db.collection('articleSources').remove();
  });

  it('blocks non-logged in users', async () => {
    const { errors } = await gql`
      mutation($article: ArticleInput!) {
        createArticle(article: $article) {
          id
          text
          paragraphs {
            text
          }
        }
      }
    `({
      article: { text: 'hahahha' },
    });

    // Should be unauthorzied
    expect(errors).toMatchSnapshot();
  });
});
