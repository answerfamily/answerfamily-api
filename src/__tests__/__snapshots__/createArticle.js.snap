// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createArticle blocks non-logged in users 1`] = `
Array [
  [GraphQLError: Must log in first],
]
`;

exports[`createArticle connects reply and paragraph 1`] = `
Object {
  "paragraphs": Array [
    Object {
      "text": "新聞",
      "user": Object {
        "id": "user-id",
      },
    },
  ],
  "sources": Array [
    Object {
      "note": "真的",
      "url": "http://google.com",
      "user": Object {
        "id": "user-id",
      },
    },
  ],
  "text": "新文章ㄛ",
  "user": Object {
    "id": "user-id",
  },
}
`;

exports[`createArticle connects reply and paragraph: scrapUrl calls 1`] = `
Array [
  Array [
    "新文章ㄛ",
    Object {
      "cacheLoader": DataLoader {
        "_batchLoadFn": [Function],
        "_options": undefined,
        "_promiseCache": Map {},
        "_queue": Array [],
      },
      "client": Promise {},
    },
  ],
  Array [
    "http://google.com",
    Object {
      "cacheLoader": DataLoader {
        "_batchLoadFn": [Function],
        "_options": undefined,
        "_promiseCache": Map {},
        "_queue": Array [],
      },
      "client": Promise {},
    },
  ],
]
`;
